django/test/runner.py

```diff
@@ -41,6 +50,119 @@
 except ImportError:
     tblib = None
 
+def recursive_injection(callstack):
+  body2 = []
+
+  for b in callstack.body:
+    #print(b, end="Next, \n")
+    bodyExists = True
+    try:
+      b.body
+    except:
+      bodyExists = False
+
+    if bodyExists:
+      recursive_injection(b)
+    
+    
+    inject_func = ast.parse(f"state_output(locals(), \"{b}\")")
+    body2.append(inject_func)
+    body2.append(b)
+
+    #print(body2)
+  
+  body2.append(inject_func)
```

Comment:
There's a potential bug here:
- if `callstack.body == []`, then inject_func is not defined
- Adding a comment about why you inject 2x might help improve understanding :)